File: intentpassing2/app/src/main/java/com/example/intentpassing2/MainActivity2.java
package com.example.intentpassing2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity2 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        Intent fromact=getIntent();




       String name1=fromact.getStringExtra("Name1");

       int rollno2=fromact.getIntExtra("Rollno1",0);



        String title=fromact.getStringExtra("title");


        String name=fromact.getStringExtra("name");


       int rollno= fromact.getIntExtra("Rollno",0);

        TextView text;

        text=findViewById(R.id.text);

      text.setText("Name"+name+"Roll no"+rollno);



        text.setText("Name"+name1+"rollno"+rollno2);


      getSupportActionBar().setTitle(title);




    }
}

File: intent/app/src/main/java/com/example/intent/MainActivity2.kt
package com.example.intent

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import com.example.intent.databinding.ActivityMainBinding
import java.util.zip.Inflater

class MainActivity2 : AppCompatActivity() {
    private val binding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)


    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {

        menuInflater.inflate(R.menu.menu,menu)

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.setting-> Toast.makeText(this@MainActivity2, "setting", Toast.LENGTH_SHORT).show()
            R.id.hello-> Toast.makeText(this@MainActivity2, "hello", Toast.LENGTH_SHORT).show()
            R.id.logout-> Toast.makeText(this@MainActivity2, "logout", Toast.LENGTH_SHORT).show()



        }



        return true
    }
}